test/lib/templateFactory.test.js

```diff
@@ -1249,6 +1331,164 @@ describe('Template Factory', () => {
+                    },
+                    lastRun: '2023-08-31T18:18:37.501Z',
+                    admins: { nathom: true }
+                }
+            ];
+
+            jobFactoryTestOutput = [
+                { pipelineId: 4, count: 1 },
+                { pipelineId: 5, count: 2 },
+                { pipelineId: 6, count: 1 }
+            ];
+
+            pipelineFactoryTestOutput = [
+                {
+                    id: 6,
+                    name: 'nathom/sd-uses-template',
+                    scmUri: 'github.com:672032066:main:pipe1',
+                    scmContext: 'github:github.com',
+                    scmRepo: {
+                        branch: 'main',
+                        name: 'test/repo',
+                        url: 'https://github.com/test/repo/tree/main/pipe1',
+                        rootDir: 'pipe1',
+                        private: false
+                    },
+                    createTime: '2023-08-17T18:18:37.501Z',
+                    admins: { nathom: true },
+                    lastEventId: 2
+                },
+
+                {
+                    id: 5,
+                    name: 'nathom/sd-uses-template',
+                    scmUri: 'github.com:672032066:main:pipe1',
+                    scmContext: 'github:github.com',
+                    scmRepo: {
+                        branch: 'main',
+                        name: 'test/repo',
+                        url: 'https://github.com/test/repo/tree/main/pipe1',
+                        rootDir: 'pipe1',
+                        private: false
+                    },
+                    createTime: '2023-08-17T18:18:37.501Z',
+                    admins: { nathom: true },
+                    lastEventId: null
+                },
+
+                {
+                    id: 4,
+                    name: 'nathom/sd-uses-template',
+                    scmUri: 'github.com:672032066:main:pipe1',
+                    scmContext: 'github:github.com',
+                    scmRepo: {
+                        branch: 'main',
+                        name: 'test/repo',
+                        url: 'https://github.com/test/repo/tree/main/pipe1',
+                        rootDir: 'pipe1',
+                        private: false
+                    },
+                    createTime: '2023-08-17T18:18:37.501Z',
+                    admins: { nathom: true },
+                    lastEventId: 1
+                }
+            ];
+
+            eventFactoryTestOutput = [
+                {
+                    id: 1,
+                    createTime: '2023-08-31T18:18:37.501Z'
+                },
+                {
+                    id: 2,
+                    createTime: '2023-08-17T18:18:37.501Z'
+                }
+            ];
+        });
+
+        it('should return empty array when no pipelines are using the template', () => {
+            const expected = [];
+
+            datastore.scan.onCall(0).resolves([]);
```

Comment:
`jobFactoryMock.list`, `pipelineFactoryMock.list` and `eventFactoryMock.list` are mocked explicitly. Do we need to mock `datastore.scan`? If not please remove it from all the tests
```suggestion
            datastore.scan.onCall(0).resolves([]);
```

Reply:
Yes, it's for mocking the initial getTemplate call.